### PR TITLE
Add force flag for overwriting licenses manually/adding licenses that is not listed in build.gradle

### LINKED ISSUE #78: Feature Request: feature for adding/changing licenses by hand
I used a library in NDK that is not listed in build.gradle. And now, I want to add the license to yaml file but received following error:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkLicenses'.
> 1

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':app:checkLicenses'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:100)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:70)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:63)
        at org.gradle.api.internal.tasks.execution.ResolveTaskOutputCachingStateExecuter.execute(ResolveTaskOutputCachingStateExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:88)
        at org.gradle.api.internal.tasks.execution.ResolveTaskArtifactStateTaskExecuter.execute(ResolveTaskArtifactStateTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:54)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.api.internal.tasks.execution.CatchExceptionTaskExecuter.execute(CatchExceptionTaskExecuter.java:34)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker$1.run(DefaultTaskGraphExecuter.java:248)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:241)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:230)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.processTask(DefaultTaskPlanExecutor.java:124)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.access$200(DefaultTaskPlanExecutor.java:80)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:105)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker$1.execute(DefaultTaskPlanExecutor.java:99)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.execute(DefaultTaskExecutionPlan.java:625)
        at org.gradle.execution.taskgraph.DefaultTaskExecutionPlan.executeWithTask(DefaultTaskExecutionPlan.java:580)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor$TaskExecutorWorker.run(DefaultTaskPlanExecutor.java:99)
        at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:63)
        at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:46)
        at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:55)
Caused by: java.lang.ArrayIndexOutOfBoundsException: 1
        at com.cookpad.android.licensetools.ArtifactId.parse(ArtifactId.java:24)
        at com.cookpad.android.licensetools.LibraryInfo.fromYaml(LibraryInfo.groovy:28)
        at com.cookpad.android.licensetools.LibraryInfo$fromYaml.call(Unknown Source)
        at com.cookpad.android.licensetools.LicenseToolsPlugin.loadLibrariesYaml(LicenseToolsPlugin.groovy:105)
        at com.cookpad.android.licensetools.LicenseToolsPlugin$loadLibrariesYaml.callCurrent(Unknown Source)
        at com.cookpad.android.licensetools.LicenseToolsPlugin.initialize(LicenseToolsPlugin.groovy:94)
        at com.cookpad.android.licensetools.LicenseToolsPlugin$_apply_closure1.doCall(LicenseToolsPlugin.groovy:27)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:681)
        at org.gradle.api.internal.AbstractTask$ClosureTaskAction.execute(AbstractTask.java:656)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter$1.run(ExecuteActionsTaskExecuter.java:122)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:336)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor$RunnableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:328)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:197)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:107)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:111)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:92)
        ... 27 more
```

I guess this is because this plugin cannot find the artifactId in build.gradle file. 

Now, I want to add license that is not listed in build.gradle. It's impossible to cover all possible ways to import library, right? So, it would be really helpful that we can add licenses by hand and generate HTML based on the information. 

Related: https://twitter.com/chibatching/status/954232965223014401

### PR DESCRIPTION
**Pull request by shiraji**
As I commented https://github.com/cookpad/license-tools-plugin/issues/78#issuecomment-359734235, I'm not able to run `./gradlew generateLicensesJson` to test whether generating JSON works or not. 

Close #78

### CODE REVIEW

**Comment by gfx** [comment]
Sounds good :thumbsup:

File: example/licenses.yml [hunk 1]
```diff
@@ -62,3 +62,36 @@
...
+  copyrightHolder: Twitter, Inc.
+  license: The Apache Software License, Version 2.0
+  licenseUrl: http://www.apache.org/licenses/LICENSE-2.0.txt
+  url: https://github.com/twitter/twitter-text
+- artifact: com.twitter.sdk.android:+:+
+  name: Tweet Ui Android SDK
+  copyrightHolder: Twitter, Inc.
+  license: bsd_3_clauses
+  licenseUrl: https://dev.twitter.com/overview/terms/twitterkit.html
+  force: true
```
Comment (ghost) [inline example/licenses.yml]: Can you rename the flag to `forceGenerate` to make sure what the flag do here?

**Review by ghost — CHANGES_REQUESTED** [review]
Except for the naming, looks great. Thanks!
We'll append the explanation in README after merging this.

File: example/licenses.yml (hunk 1 above)
Comment (shiraji) [reply example/licenses.yml]: Sure

Comment (shiraji) [reply example/licenses.yml]: Done and squashed the commits

File: plugin/src/main/groovy/com/cookpad/android/licensetools/DependencySet.java [hunk 2]
```diff
@@ -73,7 +73,7 @@ public DependencySet licensesNotMatched(DependencySet librariesYaml) {
             }
 
             for (LibraryInfo b : findAll(a.getArtifactId())) {
-                if (b.isSkip()) {
+                if (b.isSkip() || a.isForceGenerate()) {
```
Comment (slightair) [inline plugin/src/main/groovy/com/cookpad/android/licensetools/DependencySet.java]: `b.isForceGenerate()` ?

Comment (shiraji) [reply plugin/src/main/groovy/com/cookpad/android/licensetools/DependencySet.java]: nice catch

Comment (shiraji) [reply plugin/src/main/groovy/com/cookpad/android/licensetools/DependencySet.java]: Not sure how I can test this, thou.

Comment (shiraji) [reply plugin/src/main/groovy/com/cookpad/android/licensetools/DependencySet.java]: done/squashed the commits

**Review by ghost — APPROVED** [review]

**Review by slightair — APPROVED** [review]
Thanks! 😸

**Pull request merged**